spec/records_spec.rb

```diff
@@ -0,0 +1,254 @@
+      end
+    end
+  end
+
+  describe ".load_from_ndj" do
+    it "does not raise" do
+      fixture = File.join(FIXTURES_DIR, TEST_NDJ_FILE)
+      logger = NullLogger.new($stdout)
+      expect { described_class.load_from_ndj(fixture, logger: logger) }.not_to raise_error
+    end
+  end
+
+  describe "#dump_to_ndj" do
+    it "creates readable dump file" do
+      recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+      recs.dump_to_ndj TEST_NDJ_FILE
+      expect(File.readable?(TEST_NDJ_FILE)).to eq true
+    end
+  end
+
+  describe "#current_record_for" do
+    it "is initially empty" do
+      recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+      expect(recs.current_record_for(TEST_HTID)).to be_nil
+    end
+
+    it "is is populated by #compute_current_sets!" do
+      recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+      recs.logger = nil
+      recs.compute_current_sets! TEST_YYYYMM
+      expect(recs.current_record_for(TEST_HTID)).not_to be_nil
+    end
+  end
+
+  describe "#redirects" do
+    it "returns non-empty Hash of redirects" do
+      recs = records
+      recs.add_monthly File.join(FIXTURES_DIR, TEST_OLDER_SAMPLE_HATHIFILE_NAME)
+      recs.add_monthly File.join(FIXTURES_DIR, TEST_SAMPLE_HATHIFILE_NAME)
+      recs.compute_current_sets!(recs.newest_load)
+      recs.remove_dead_htids!
+      redirects = recs.redirects
+      expect(redirects).not_to be_empty
+      expect(redirects.keys.all? { |x| x.instance_of?(Integer) }).to eq true
+      expect(redirects.values.all? { |x| x.instance_of?(Integer) }).to eq true
+      # puts "REDIRECTS: #{recs.redirects.inspect}"
+    end
+  end
+
+  describe "#remove_dead_htids!" do
+    context "without previous call to #compute_current_sets!" do
+      it "raises" do
+        recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+        expect { recs.remove_dead_htids! }.to raise_error StandardError
+      end
+    end
+
+    context "with previous call to #compute_current_sets!" do
+      it "does not raise" do
+        recs = records.add(htid: TEST_HTID, recid: TEST_RECID, yyyymm: TEST_YYYYMM)
+        recs.compute_current_sets!
+        expect { recs.remove_dead_htids! }.not_to raise_error
+      end
+    end
+  end
+
+  describe "#intify_record_id" do
+    it "removes leading zeroes and returns Integer" do
+      cases = {"0000001" => 1, "1" => 1, "1000000" => 1000000, "nonsense" => 0}
+      cases.each_pair do |record_id, int|
+        expect(records.intify_record_id(record_id)).to eq int
+      end
+    end
+  end
+
+  describe ".yyyymm_from_filename" do
+    context "with a valid filename" do
+      it "returns YYYYMM" do
+        expect(described_class.yyyymm_from_filename(TEST_VALID_HATHIFILE_NAME)).to eq TEST_YYYYMM.to_i
+      end
+    end
+
+    context "with an invalid filename" do
+      it "raises" do
+        # Doesn't look like it ever actually hits exit(1)
```

Comment:
This was confusing. I am certain `yyyymm_from_filename` can be uncomplicated, and that will simplify associated tests.

Reply:
Like you point out in comments in some of these tests, `# May need a rewrite`

Reply:
I mistakenly left my comments in the spec as TODO notes. It does successfully call `exit(1)` which is kinda sketchy behavior in general, however the class method is a useful utility for scripts in `bin/` so I can excuse it.